conan/tools/gnu/makedeps.py

```diff
@@ -0,0 +1,588 @@
+        :param deps_cpp_info_dirs: Formatted dependencies folders
+        :param deps_cpp_info_flags: Formatted dependencies variables
+        """
+        context = {"dirs": deps_cpp_info_dirs, "flags": deps_cpp_info_flags}
+        template = Template(self.template, trim_blocks=True, lstrip_blocks=True, undefined=StrictUndefined)
+        return template.render(context)
+
+    def deps_content(self, dependencies_names: list) -> str:
+        """
+        Generate content for CONAN_DEPS (e.g. CONAN_DEPS = zlib, openssl)
+        :param dependencies_names: Non-formatted dependencies names
+        """
+        context = {"deps": dependencies_names}
+        template = Template(self.template_deps, trim_blocks=True, lstrip_blocks=True, undefined=StrictUndefined)
+        return template.render(context)
+
+
+class GlobalGenerator:
+    """
+    Process all collected dependencies and parse to generate global content
+    """
+
+    def __init__(self, conanfile, make_infos):
+        self._conanfile = conanfile
+        self._make_infos = make_infos
+
+    def _get_dependency_dirs(self) -> dict:
+        """
+        List regular directories from cpp_info and format them to be used in the makefile
+        """
+        dirs = {}
+        for var in _common_cppinfo_dirs():
+            key = var.replace("dirs", "_dirs")
+            dirs[key] = [f"$(CONAN_{key.upper()}_{_makefy(makeinfo.name)})" for makeinfo in self._make_infos if var in makeinfo.dirs]
+        return dirs
+
+    def _get_dependency_flags(self) -> dict:
+        """
+        List common variables from cpp_info and format them to be used in the makefile
+        """
+        flags = {}
+        for var in _common_cppinfo_variables():
+            key = var.replace("dirs", "_dirs")
+            flags[key] = [f"$(CONAN_{key.upper()}_{_makefy(makeinfo.name)})" for makeinfo in self._make_infos if var in makeinfo.flags]
+        return flags
+
+    def generate(self) -> str:
+        """
+        Process folder and variables for a dependency and generates its Makefile content
+        """
+        glob_content_gen = GlobalContentGenerator()
+        dirs = self._get_dependency_dirs()
+        flags = self._get_dependency_flags()
+        return glob_content_gen.content(dirs, flags)
+
+    def deps_generate(self) -> str:
+        """
+        Process dependencies names and generates its Makefile content.
+        It should be added as first variable in the Makefile.
+        """
+        dependencies = [makeinfo.name for makeinfo in self._make_infos if makeinfo.name != self._conanfile.name]
+        glob_content_gen = GlobalContentGenerator()
+        return glob_content_gen.deps_content(dependencies)
+
+
+class DepComponentContentGenerator:
+    """
+    Generates Makefile content for each dependency component
+    """
+
+    template = textwrap.dedent("""\
+        {%- macro format_map_values(values) -%}
```

Comment:
When recommending to use more jinja templates, I was thinking of being able to visualize a bit better the actual contents, like in https://github.com/conan-io/conan/blob/c53f2879f5ca6d7324455c8fcf9ea4ab8e358ab9/conan/tools/cmake/cmakedeps/templates/target_data.py#L112
```cmake
set({{ pkg_name }}_INCLUDE_DIRS{{ config_suffix }} {{ global_cpp.include_paths }})
set({{ pkg_name }}_RES_DIRS{{ config_suffix }} {{ global_cpp.res_paths }})
set({{ pkg_name }}_DEFINITIONS{{ config_suffix }} {{ global_cpp.defines }})
set({{ pkg_name }}_SHARED_LINK_FLAGS{{ config_suffix }} {{ global_cpp.sharedlinkflags_list }})
```

I am not sure this template helps much in understanding the actual file contents, maybe it is because still too much abstraction over the dirs and flags.

Reply:
It's possible to be more explicit, but it would result in multiple ifs as we don't to expose unused variables, otherwise the final conandeps.mk would be polluted with multiple empty variables. Let me update to a more explicit way, we still can revert in case we don't like.

Reply:
@memsharded Please, take a look again. Now variables are explicit in the template. I moved the method that formats lines, otherwise we would need to replicate 3 times. It's possible to avoid even more duplication, but I guess on this current state is more readable.